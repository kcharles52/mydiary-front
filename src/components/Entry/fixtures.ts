export const props = {
  entry_id: 1,
  title: "Title 1",
  diaryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete:jest.fn()

};
