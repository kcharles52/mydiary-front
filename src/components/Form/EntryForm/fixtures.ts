export const EntryFormProps = {
  Edit: false,
  title: "string",
  diaryTitle: "string",
  date: "string",
  diaryBody: "string",
  onSubmit: jest.fn(),
  onCancel: jest.fn(),
  onChange: jest.fn()
};

  