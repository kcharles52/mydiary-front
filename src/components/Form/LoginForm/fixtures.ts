export const LoginFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  