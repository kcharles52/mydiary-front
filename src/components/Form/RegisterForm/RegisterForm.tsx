//react libraries
import * as React from "react";

//components
import { Input, Button } from "../common";

//interfaces
import { RegisterProps } from "./interfaces";
export const RegisterForm: React.SFC<RegisterProps> = props => {
  return (
    <form action="">
      <h1>Register for an account</h1>
      <Input
        name="name"
        type="text"
        label="Full Name *"
        value={props.name ||""}
        onChange={props.onChange}
      />
      <Input
        type="text"
        name="email"
        label="Email Address *"
        value={props.email||""}
        onChange={props.onChange}
      />
      <Input
        type="password"
        name="password"
        label="Password *"
        value={props.password||""}
        onChange={props.onChange}
      />
      <Input
        type="password"
        name="ConfirmPassword"
        label="Confirm Password *"
        value={props.ConfirmPassword||""}
        onChange={props.onChange}
      />
      <Button
        label="Get started"
        className="button button-block"
        onClick={()=>{props.onSubmit("signUp")}}
      />
    </form>
  );
};
