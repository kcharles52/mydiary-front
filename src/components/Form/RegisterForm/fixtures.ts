export const RegisterFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  