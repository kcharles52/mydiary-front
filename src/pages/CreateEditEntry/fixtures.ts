export const props = {
  Edit: false,
  title: "Form title",
  diaryTitle: "Test Diary Title",
  date: "20-2-19",
  diaryBody: "Test Diary body",
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  addEntryAction: jest.fn(),
  editEntryAction: jest.fn(),
  getEntryAction: jest.fn(() =>
    Promise.resolve({
      entry: {
        entry_id: 1,
        diaryTitle: "Title 1",
        diaryEntryBody:
          "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
        date: "12/12/2018",
        delete: jest.fn()
      }
    })
  ),
  match: { params: { entry_id: "1" } },
  editEntry: {
    entry: {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  },
  newEntry: {},
  history: { push: () => {} },
  location: { pathname: "/addEntry" }
};

export const entry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
