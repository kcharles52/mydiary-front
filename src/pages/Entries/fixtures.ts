export const props = {
  getEntriesAction: jest.fn(),
  deleteEntryAction: jest.fn(),
  entries: [
    {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      diaryTitle: "Title 2",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  deleted: [],
  handlesDelete: jest.fn(),
  history: {push:jest.fn()},
  location: { pathname: "/home" },
  match: { url: "/home" }
};

export const fetchedEntries =[
  {
    entry_id: 1,
    diaryTitle: "Title 1",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  },
  {
    entry_id: 2,
    diaryTitle: "Title 2",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  }
]