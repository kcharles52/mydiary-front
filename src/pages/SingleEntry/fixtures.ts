export const props = {
  getEntryAction: (id: string) => Promise.resolve(),
  deleteEntryAction: (id: string) => Promise.resolve(),
  fetchedEntry: { entry: {} },
  match: { params: { entry_id: "1" } },
  history: { push: jest.fn() },
  location: { pathname: "/Entry/1" }
};

export const fetchedEntry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
