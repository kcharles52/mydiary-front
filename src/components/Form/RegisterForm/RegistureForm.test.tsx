// react libraries
import * as React from "react";

// third party libraries
import { mount, shallow } from "enzyme";

// components
import { RegisterForm } from "./RegisterForm";

//fixtures
import { RegisterFormProps as props } from "./fixtures";

describe("Register form componet", () => {
  let wrapper, shallowWrapper;
  beforeEach(() => {
    wrapper = mount(<RegisterForm {...props} />);
    shallowWrapper = shallow(<RegisterForm {...props} />);
  });
  afterEach(() => {
    wrapper = shallowWrapper = null;
  });

  it("Should render register form componet", () => {
    expect(shallowWrapper).toMatchSnapshot();
    expect(shallowWrapper.find("Input").length).toBe(4);
    expect(shallowWrapper.find("Button").length).toBe(1);
  });
  it("should register user", () => {
    wrapper.find("Button").simulate("click");
    expect(wrapper.prop("onSubmit")).toBeCalled();
  });
});
