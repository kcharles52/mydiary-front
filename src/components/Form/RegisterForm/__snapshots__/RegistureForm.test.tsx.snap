// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register form componet Should render register form componet 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegisterForm
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "action": "",
      "children": Array [
        <h1>
          Register for an account
        </h1>,
        <Input
          label="Full Name *"
          name="name"
          onChange={[MockFunction]}
          type="text"
          value=""
        />,
        <Input
          label="Email Address *"
          name="email"
          onChange={[MockFunction]}
          type="text"
          value=""
        />,
        <Input
          label="Password *"
          name="password"
          onChange={[MockFunction]}
          type="password"
          value=""
        />,
        <Input
          label="Confirm Password *"
          name="ConfirmPassword"
          onChange={[MockFunction]}
          type="password"
          value=""
        />,
        <Button
          className="button button-block"
          label="Get started"
          onClick={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Register for an account",
        },
        "ref": null,
        "rendered": "Register for an account",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Full Name *",
          "name": "name",
          "onChange": [MockFunction],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Email Address *",
          "name": "email",
          "onChange": [MockFunction],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Password *",
          "name": "password",
          "onChange": [MockFunction],
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "label": "Confirm Password *",
          "name": "ConfirmPassword",
          "onChange": [MockFunction],
          "type": "password",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "button button-block",
          "label": "Get started",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "",
        "children": Array [
          <h1>
            Register for an account
          </h1>,
          <Input
            label="Full Name *"
            name="name"
            onChange={[MockFunction]}
            type="text"
            value=""
          />,
          <Input
            label="Email Address *"
            name="email"
            onChange={[MockFunction]}
            type="text"
            value=""
          />,
          <Input
            label="Password *"
            name="password"
            onChange={[MockFunction]}
            type="password"
            value=""
          />,
          <Input
            label="Confirm Password *"
            name="ConfirmPassword"
            onChange={[MockFunction]}
            type="password"
            value=""
          />,
          <Button
            className="button button-block"
            label="Get started"
            onClick={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Register for an account",
          },
          "ref": null,
          "rendered": "Register for an account",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Full Name *",
            "name": "name",
            "onChange": [MockFunction],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Email Address *",
            "name": "email",
            "onChange": [MockFunction],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Password *",
            "name": "password",
            "onChange": [MockFunction],
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "label": "Confirm Password *",
            "name": "ConfirmPassword",
            "onChange": [MockFunction],
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "button button-block",
            "label": "Get started",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
